src/qmapshack/poi/CRawPoi.cpp

```diff
@@ -19,63 +19,363 @@
+
+    //Dear translators, please reformat this the way you want which is appropriate for your locale. The osm key will be replaced by the value.
+    //Note that a non-letter character is needed between a key and another key or some arbitrary letter that is part of the address scheme
+    //You can find all avalable address keys at: https://wiki.openstreetmap.org/wiki/Key:addr
+    QString addr = tr("addr:housenumber addr:street<br>\naddr:city addr:postcode<br>\naddr:country");
+    static const QRegularExpression addressRegex("(addr:[A-z]+)", QRegularExpression::MultilineOption);
+    QRegularExpressionMatchIterator i = addressRegex.globalMatch(addr);
+    while (i.hasNext())
     {
-        return category;
+        const QRegularExpressionMatch& match = i.next();
+        const QString& key = match.captured(1);
+        if(data.contains(key))
+        {
+            addr.replace(key, data[key]);
+            tempSkipKeys.append(key);
+        }
+        else
+        {
+            addr.replace(key, "");
+        }
     }
-    return name;
+    static const QRegularExpression addrCleanup("(.*?)(\\s*<br>\n\\s*)+\\Z", QRegularExpression::MultilineOption | QRegularExpression::DotMatchesEverythingOption);
+    QRegularExpressionMatch match = addrCleanup.match(addr);
+    if(match.hasMatch())
+    {
+        addr = match.captured(1);
+    }
+    desc += addr;
+
+    if(nameRelatedKeys.count() > 1)
+    {
+        if(!desc.isEmpty())
+        {
+            desc += "<br>\n";
+        }
+        desc += "<b>" + tr("Alternative names: ") + "</b>";
+        bool firstName = true;
+        for(const QString& key : nameRelatedKeys)
+        {
+            if(data[key] == name)
+            {
+                continue;
+            }
+
+            if(firstName)
+            {
+                firstName = false;
+            }
+            else
+            {
+                desc += ", ";
+            }
+            desc += data[key];
+        }
+    }
+    if(data.contains("phone"))
+    {
+        if(!desc.isEmpty())
+        {
+            desc += "<br>\n";
+        }
+        desc += "<b>" + tr("Phone number: ") + "</b>" + data["phone"];
+    }
+    if(data.contains("contact:phone"))
+    {
+        if(!desc.isEmpty())
+        {
+            desc += "<br>\n";
+        }
+        desc += "<b>" + tr("Phone number: ") + "</b>" + data["contact:phone"];
+    }
+    if(data.contains("contact:mobile"))
+    {
+        if(!desc.isEmpty())
+        {
+            desc += "<br>\n";
+        }
+        desc += "<b>" + tr("Mobile phone number: ") + "</b>" + data["contact:mobile"];
+    }
+    if(data.contains("fax"))
+    {
+        if(!desc.isEmpty())
+        {
+            desc += "<br>\n";
+        }
+        desc += "<b>" + tr("Fax: ") + "</b>" + data["fax"];
+    }
+    if(!desc.isEmpty())
+    {
+        desc += "<br>\n" "<br>\n";
+    }
+    // Append all other tags
+    bool firstTag = true;
+    for(const QString& key : data)
```

Comment:
I think there is another bug. It should read 
```
    const QStringList& keys = data.keys();
    for(const QString& key : keys)
```
 Could you please check all other locations where you used range loops, too. And actually test you code before you commit ;)

Reply:
Sorry. I tested it better now and found a segmentation fault when selecting POIs from the map but can't figure out it's cause.
It happens right after you selected whether you want to choose a symbol manually or not when you prompt it to copy the selected items over to a project, regardless of your response. It however either faults in the `SETTINGS` macro if you select `No` and in `QString::isEmpty()` if you select `yes`. The mutex locker is there, so that's not the cause and the pois that should be copied look intact. Do you have an idea what's the cause?

Reply:
start with gdb and analyze the backtrace?

Reply:
Btw we still have issus #357, too. Seems to be only on the POI feature branch. Probably some mutex problem.

Reply:
Just to keep track: The segfault you mentioned is not fixed, yet. Issue #357 is not reproducible. Correct?

Reply:
Bump! Any news on that?

Reply:
Didn't have the nerves yet to to investigate the reason, as I currently code much for work.
But I think I will spend some time on it this weekend, as it was a short working week :wink:

Reply:
> Just to keep track: The segfault you mentioned is not fixed, yet. Issue #357 is not reproducible. Correct?

Correct. 

Could you please check if you get the segfault when you try to select items on the map and copy them to a project? I think I have it traced down to the `exec()`calls of the dialogs that spin an additional event loop after which it faults when accessing the `poi_t`that it currently copies. Since it is only after the dialogs that it fails, you have to select an item that has the default POI symbol. Since it is something that definitely worked before, I'd like to check if it might have to do with my flawed libgdal setup (although I doubt it).

Reply:
I couldn't reproduce the crash. However I noticed another bug. You should move the line 

```
wpt->setReadOnlyMode(true);
```

behind `wpt->editInitial();`  else the icon is not editable (unless you unlock the item first)

I just wonder if the crash also occurs when you run QMS with gdb. Or does it produce a core dump that can be analyzed. Are there any Qt related messages when running with `-d`?

Reply:
> I couldn't reproduce the crash. However I noticed another bug. You should move the line
> 
> ```
> wpt->setReadOnlyMode(true);
> ```
> 
> behind `wpt->editInitial();` else the icon is not editable (unless you unlock the item first)

I'll do that :+1: 

> Are there any Qt related messages when running with -d?

There were, but that was related to another edge-case bug that occurred by chance which I managed to fix :laughing: 
I used `qDebug()` to trace down the issue to the `exec()` as `qDebug() << "icon: " << poi.icon;` works before doing `exec()` on the dialog window but not after (at least for me).